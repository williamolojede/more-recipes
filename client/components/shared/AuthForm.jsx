import React from 'react';
import PropTypes from 'prop-types';

/**
 * @class AuthForm
 * @extends {Component}
 */
class AuthForm extends React.Component {
  /**
   * @param {node} e - element
   * @return {view} - renders a form
   */
  onFormSubmit(e) {
    e.preventDefault();

    const email = this.emailInput.value;
    const password = this.passwordInput.value;
    if (this.props.type === 'login') {
      this.props.authFormSubmit({ email, password });
    }

    this.authForm.reset();
  }

  /**
   * @return {view} - renders a form
   */
  render() {
    return (
      <form
        className="auth-form form"
        ref={(node) => { this.authForm = node; }}
        onSubmit={e => this.onFormSubmit(e)}
      >
        {
          this.props.type === 'signup' ?
            <input
              type="text"
              name="usr-name"
              className="form__input"
              placeholder="name"
              required
            />
            :
            null
        }
        <input
          id="usr-email"
          type="email"
          name="usr-email"
          className="form__input"
          placeholder="email"
          required
          ref={(node) => { this.emailInput = node; }}
        />
        <input
          id="usr-pswd"
          type="password"
          name="usr-pswd"
          className="form__input"
          placeholder="password"
          required
          ref={(node) => { this.passwordInput = node; }}
        />

        <button type="submit" className="form__submit z-depth-1">
          {
            this.props.type === 'signup' ?
              'Register'
              :
              'Login'
          }
        </button>
      </form>
    );
  }
}

AuthForm.propTypes = {
  type: PropTypes.string.isRequired,
  authFormSubmit: PropTypes.func.isRequired,
};

export default AuthForm;
