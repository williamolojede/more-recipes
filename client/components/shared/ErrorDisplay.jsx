import React from 'react';
import { errorDisplayPropTypes } from '../../config/proptypes';

const ErrorDisplay = props => (
  <div className="error z-depth-1">
    <span className="error__cancel">X</span>
    <p className="error__message">{ props.message }</p>
  </div>
);

ErrorDisplay.propTypes = errorDisplayPropTypes;

export default ErrorDisplay;
