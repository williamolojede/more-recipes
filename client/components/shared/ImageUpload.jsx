import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ReactLoading from 'react-loading';

class ImageUpload extends Component {
  render() {
    const {
      imgUrl,
      altText,
      imageUploadProgress,
      className,
      defaultImgUrl,
      imageUploadHandler
    } = this.props;

    return (
      <div className={`image-upload ${className}`}>
        <div className="image-upload__preview">
          <img
            src={imgUrl || defaultImgUrl}
            alt={altText}
            className="image-upload__preview___image"
          />
          <div className="image-upload__progress">
            {
              imageUploadProgress === 'started' &&
              <ReactLoading
                type="spokes"
                color="#000"
                height="65px"
                width="65px"
                className="loading-icon"
              />
            }
          </div>
        </div>
        <label htmlFor="recipe-image" className="z-depth-1">Upload new picture</label>
        <input
          type="file"
          id="recipe-image"
          name="filename"
          accept="image/gif, image/jpeg, image/png"
          ref={node => this.recipeImageInput = node}
          onChange={() => imageUploadHandler(this.recipeImageInput)}
        />
      </div>
    );
  }
}

ImageUpload.defaultProps = {
  className: '',
  imgUrl: ''
};

ImageUpload.propTypes = {
  className: PropTypes.string,
  imgUrl: PropTypes.string,
  altText: PropTypes.string.isRequired,
  defaultImgUrl: PropTypes.string.isRequired,
  imageUploadProgress: PropTypes.oneOf(['pending', 'started', 'failed', 'completed']).isRequired,
  imageUploadHandler: PropTypes.func.isRequired
};

export default ImageUpload;
