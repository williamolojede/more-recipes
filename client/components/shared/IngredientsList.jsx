import React from 'react';

const IngredientsList = () => (
  <div className="recipe__ingredients col s12 m5">
    <h3>Ingredients</h3>
    <ul>
      <li><span>1 1/2 cups</span> all-purpose flour</li>
      <li><span>1 1/4 cups</span> milk</li>
      <li><span>3 1/2 teaspoons</span> baking powder</li>
      <li><span>1</span> egg</li>
      <li><span>1 teaspoon</span> salt</li>
      <li><span>3 tablespoons</span> butter, melted</li>
      <li><span>1 tablespoon </span> white sugar</li>
    </ul>
  </div>
);

export default IngredientsList;
