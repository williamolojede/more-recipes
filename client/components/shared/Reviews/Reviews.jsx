import React from 'react';
import PropTypes from 'prop-types';

import { userPrototypes } from '../../../config/proptypes';

import Review from './Review.jsx';
import AddReviewForm from './AddReviewForm.jsx';

const Reviews = ({ reviews, user, id }) => (
  <div className="recipe__reviews row">
    <h3>Reviews</h3>
    <ul className="review__list">
      {
        reviews.length === 0 ?
          <li className="no-reviews" >eyah <span role="img" aria-label="sad">ðŸ˜¢  </span> no reviews yet</li>
          :
          reviews.map((review, i) => <Review review={review} key={i} />)
      }
      <AddReviewForm user={user} id={id} />
    </ul>
  </div>
);

Reviews.propTypes = {
  user: PropTypes.shape(userPrototypes).isRequired
};

export default Reviews;
