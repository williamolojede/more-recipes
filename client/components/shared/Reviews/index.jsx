/* eslint react/no-array-index-key: 0 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ReactPaginate from 'react-paginate';

import { fetchReviews } from '../../../actions/reviews';
import {
  reviewPropTypes,
  currentUserPropTypes,
  paginationPropTypes
} from '../../../config/proptypes';

import Review from './Review.jsx';
import AddReviewForm from './AddReviewForm.jsx';
import Preloader from '../Preloader';

class Reviews extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPage: 1,
      limit: 6,
      fetchingReviews: false
    };
  }

  componentDidMount() {
    const payload = {
      recipeId: this.props.recipeId,
      limit: this.state.limit,
      page: this.state.currentPage
    };
    this.props.dispatch(fetchReviews(payload));
  }


  componentWillReceiveProps(newProps) {
    if (newProps.reviews) {
      const {
        pagination
      } = newProps;
      this.setState({
        last: pagination.last
      });
    }
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.currentPage !== this.state.currentPage) {
      const payload = {
        recipeId: this.props.recipeId,
        limit: this.state.limit,
        page: this.state.currentPage
      };
      this.setState({ fetchingReviews: true });
      this.props.dispatch(fetchReviews(payload))
        .then(() => {
          this.setState({ fetchingReviews: false });
        });
    }
  }

  handlePageClick = ({ selected }) => {
    this.setState({
      currentPage: selected + 1
    });
  }


  render() {
    const { reviews, currentUser, recipeId } = this.props;
    return (
      <div className="recipe__reviews row">
        <h3>Reviews</h3>
        <AddReviewForm
          currentUser={currentUser}
          dispatch={this.props.dispatch}
          recipeId={recipeId}
          limit={this.state.limit}
        />
        <ul className="review__list">
          {
            this.state.fetchingReviews && <Preloader />
          }
          {
            reviews.length === 0 ?
              <li className="no-reviews" >eyah <span role="img" aria-label="sad">ðŸ˜¢  </span> no reviews yet</li>
              :
              reviews.map((review, i) => <Review review={review} key={i} />)
          }
        </ul>
        {
          this.state.last > 1 &&
            <ReactPaginate
              previousLabel={<i className="material-icons">chevron_left</i>}
              nextLabel={<i className="material-icons">chevron_right</i>}
              pageCount={this.props.pagination.last}
              marginPagesDisplayed={3}
              pageClassName="waves-effect"
              pageRangeDisplayed={2}
              onPageChange={this.handlePageClick}
              activeClassName="active"
              containerClassName="pagination"
            />
        }
      </div>
    );
  }
}


Reviews.propTypes = {
  reviews: PropTypes.arrayOf(reviewPropTypes.review).isRequired,
  recipeId: PropTypes.number.isRequired,
  ...currentUserPropTypes,
  ...paginationPropTypes
};

export const mapStateToProps = ({
  recipeReviews: { reviews, pagination },
  isFetching,
  auth: { currentUser }
}) => ({
  reviews,
  pagination,
  currentUser,
  isFetching
});

export { Reviews as PureReviews };

export default connect(mapStateToProps)(Reviews);
