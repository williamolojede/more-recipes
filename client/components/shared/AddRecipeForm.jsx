import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import defaultConfig from '../../config/default';
import { recipePropTypes } from '../../config/proptypes';
import {
  addRecipe,
  updateRecipe,
  fetchSingleRecipe,
  resetSingleRecipe
} from '../../actions/recipe';
import imageUpload from '../../utils/imageUpload';

import ErrorDisplay from './ErrorDisplay';
import Preloader from './Preloader';
import ImageUpload from './ImageUpload';
import SubmitButton from './SubmitButton';


class AddRecipeForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      recipe: this.props.recipe,
      renderInput: {
        ingredient: false,
        instruction: false,
      },
      submitLoading: false,
      // pending, started, failed, completed
      imageUploadProgress: 'pending',
      addRecipe: {
        allow: true,
        errorMessage: '',
      }
    };
  }

  componentDidMount() {
    if (this.props.isEditMode) {
      this.props.dispatch(fetchSingleRecipe(this.props.match.params.recipeId));
    }
  }

  componentWillReceiveProps(newProps) {
    this.setState({ recipe: newProps.recipe });
  }

  componentWillUnmount() {
    this.props.dispatch(resetSingleRecipe());
  }

  /**
   * @param {object} event  - event object
   * @param {object} el - input element
   * @param {string} type - recipe key name whose value is to be updated
   * @return {undefined}
   */
  addItem = (event, el, type) => {
    if (event.keyCode === 13) {
      event.preventDefault();
      const recipe = Object.assign({}, this.state.recipe);
      const text = el.value.trim();
      if (text) {
        recipe[type] = recipe[type].concat(text);
        this.setState({ recipe });
      }
      el.value = '';
    }
  }

  /**
   * @param {object} event  - event object
   * @param {string} type - recipe key name whose value is to be updated
   * @param {number} index - input element
   * @return {undefined}
   */
  removeItem = (event, type, index) => {
    event.preventDefault();
    const recipe = Object.assign({}, this.state.recipe);

    recipe[type] = [
      ...recipe[type].slice(0, index),
      ...recipe[type].slice(index + 1)
    ];
    this.setState({ recipe });
  }

  /**
   * Form submission handler
   * @return {undefined}
   */
  addRecipe = () => {
    const {
      name,
      description,
      ingredients,
      instructions
    } = this.state.recipe;

    if (!name || !description) {
      this.setState({
        addRecipe: {
          allow: false,
          errorMessage: 'recipe name and description are required'
        }
      });
    } else if (!ingredients.length || !instructions.length) {
      this.setState({
        addRecipe: {
          allow: false,
          errorMessage: 'ingredients and instructions cannot be empty'
        }
      });
    } else {
      this.setState({ submitLoading: true });
      this.props.dispatch(addRecipe(this.state.recipe))
        .then(this.handleFormSubmitSuccess);
    }
  }

  updateRecipe = () => {
    this.setState({ submitLoading: true });
    this.props.dispatch(updateRecipe(this.state.recipe))
      .then(this.handleFormSubmitSuccess);
  }

  handleFormSubmitSuccess = () => {
    // set submit loading to false and redirect to
    //  all recipes page after .5s
    this.setState({ submitLoading: false });
    setTimeout(() => this.props.history.push('/user/recipes'), 500);
  }

  submitForm = (e) => {
    e.preventDefault();
    if (this.props.isEditMode) {
      this.updateRecipe();
    } else {
      this.addRecipe();
    }
  }

  /**
   * @param {object} e - event object
   * @param {string} type - type of input to be show, for ingredients or instructions
   * @return {undefined}
   */
  showInput = (e, type) => {
    e.preventDefault();
    const renderInput = Object.assign({}, this.state.renderInput);

    // if already true do nothing
    if (!renderInput[type]) {
      renderInput[type] = true;
      this.setState({ renderInput });
    }
  }

  /**
   * @param {object} e - event object
   * @param {string} type - type of input to be show, for ingredients or instructions
   * @return {undefined}
   */
  hideInput = (e, type) => {
    const renderInput = Object.assign({}, this.state.renderInput);
    // if already true do nothing
    if (renderInput[type]) {
      renderInput[type] = false;
      this.setState({ renderInput });
    }
  }

  /**
   * Handles image upload
   * @param {object} imageInput - the image input to get image to be uploaded from
   * @return {undefined}
   */
  imageUploadHandler = (imageInput) => {
    imageUpload.call(
      this,
      imageInput,
      `recipes/${Date.now()}`,
      (downloadURL) => {
        const recipe = Object.assign({}, this.state.recipe);
        recipe.img_url = downloadURL;
        this.setState({ recipe, imageUploadProgress: 'completed' });
      }
    );
  }

  /**
   * Adds the recipe name or description to the state
   * @param {object} el - element
   * @param {string} type - property name whose value is to be added, type: 'name' or 'description'
   * @return {undefined}
   */
  addRecipeProp = (el, type) => {
    const text = el.value.trim();
    const recipe = Object.assign({}, this.state.recipe);
    recipe[type] = text;
    this.setState({ recipe });
  }

  /**
   * renderItems returns of a list of item that matches the provide type in state.recipe
   * @param {string} type - the type of item in state to be looped through
   * @return {jsx} - a list of items to be redered
   */
  renderItems = type => this.state.recipe[type].map((item, index) => (
    <li key={index}>
      {item}
      <a href="/delete" className={`remove-item remove-${type}`} onClick={e => this.removeItem(e, type, index)}>
        <i className="mdi mdi-delete" />
      </a>
    </li>
  ))

  renderForm = () => (
    <form
      className="my-recipe__add--form"
      onSubmit={this.submitForm}
    >
      <label className="my-recipe__add--form___name" htmlFor="recipe-name">
          Name
        <input
          type="text"
          id="recipe-name"
          name="recipe-name"
          placeholder={this.state.recipe.name}
          defaultValue={this.state.recipe.name}
          ref={node => this.recipeNameInput = node}
          onChange={() => this.addRecipeProp(this.recipeNameInput, 'name')}
        />
      </label>
      {/*  description-image wrapper  */}
      <div className="description-image">
        <label className="my-recipe__add--form___desc" htmlFor="recipe-description">
            Description
          <textarea
            placeholder="e.g: short description..."
            id="recipe-description"
            defaultValue={this.state.recipe.description}
            ref={node => this.recipeDescInput = node}
            onChange={() => this.addRecipeProp(this.recipeDescInput, 'description')}
          />
        </label>
        <div className="my-recipe__add--form___img">
          <p className="label">Image</p>
          <ImageUpload
            imgUrl={this.state.recipe.img_url}
            altText={this.state.recipe.name}
            imageUploadProgress={this.state.imageUploadProgress}
            imageUploadHandler={this.imageUploadHandler}
            defaultImgUrl={defaultConfig.recipeImgUrl}
          />
        </div>
      </div>

      {/* ingredients-instruction wrapper  */}
      <div className="ingredients-instruction">
        <fieldset className="ingredients-form my-recipe__add--form___ingredients">
          <legend>Ingredient(s):</legend>
          <ol>
            {
              this.renderItems('ingredients')
            }
          </ol>
          {
            this.state.renderInput.ingredient ?
              <textarea
                type="text"
                name="recipe-ingredient__name"
                className="ingredient-instrution-input"
                placeholder="e.g: egusi"
                ref={node => this.addIngredientInput = node}
                onKeyUp={e => this.addItem(e, this.addIngredientInput, 'ingredients')}
                onBlur={e => this.hideInput(e, 'ingredient')}
              />
              :
              <a href="/add-new" className="add-new" onClick={e => this.showInput(e, 'ingredient')}>
                <i className="mdi mdi-plus" />
                <span>Add an ingredient</span>
              </a>
          }
        </fieldset>
        <fieldset className="instructions-form my-recipe__add--form___instructions">
          <legend>Instruction(s):</legend>
          <ol>
            {
              this.renderItems('instructions')
            }
          </ol>
          {
            this.state.renderInput.instruction ?
              <textarea
                type="text"
                name="recipe-instruction__name"
                className="ingredient-instrution-input"
                placeholder="e.g: boil water"
                ref={node => this.addInstructionInput = node}
                onKeyUp={e => this.addItem(e, this.addInstructionInput, 'instructions')}
                onBlur={e => this.hideInput(e, 'instruction')}
              />
              :
              <a href="/add-new" className="add-new" onClick={e => this.showInput(e, 'instruction')}>
                <i className="mdi mdi-plus" />
                <span>Add an instruction</span>
              </a>
          }
        </fieldset>
      </div>
      <div className="form-btns">
        <SubmitButton
          submitText={this.props.isEditMode ? 'Update' : 'Submit'}
          submitLoading={this.state.submitLoading}
        />
        {
          this.props.isEditMode ?
            <Link to="/user/recipes">or go back to your recipe list</Link>
            :
            null
        }
      </div>
    </form>
  )

  render() {
    return (
      <div className="container page-manage__add z-depth-1">
        {
          !this.state.addRecipe.allow &&
            <ErrorDisplay message={this.state.addRecipe.errorMessage} />
        }
        {
          (this.props.isEditMode && this.props.isFetching) ? <Preloader />
            : this.renderForm()
        }
      </div>
    );
  }
}

AddRecipeForm.propTypes = {
  dispatch: PropTypes.func.isRequired,
  ...recipePropTypes,
  isEditMode: PropTypes.bool.isRequired,
};

const mapStateToProps = ({
  isFetching,
  singleRecipe: { recipe }
}) => ({
  isFetching,
  recipe
});

export { AddRecipeForm as PureAddRecipeForm };
export default connect(mapStateToProps)(AddRecipeForm);
