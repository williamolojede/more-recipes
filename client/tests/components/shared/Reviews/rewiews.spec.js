import React from 'react';
import { PureReviews, mapStateToProps } from '../../../../components/shared/Reviews';
import user from '../../../__mocks__/user.mock';
import allReviews from '../../../__mocks__/reviews.mock';

function setup() {
  const props = {
    reviews: allReviews.reviews,
    pagination: allReviews.pagination,
    recipeId: 2,
    currentUser: user,
    dispatch: jest.fn().mockReturnValue(Promise.resolve())
  };

  const wrapper = shallow(<PureReviews {...props} />);

  return {
    wrapper,
    props
  };
}

describe('Reviews Component', () => {
  it('should render with the right amount of elements', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a no-reviews page if there are no reviews', () => {
    const { props } = setup();
    const propsWithEmptyReviews = Object.assign({}, props, { reviews: [] });
    const wrapper = shallow(<PureReviews {...propsWithEmptyReviews} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render pagination links if last page is greater than one', () => {
    const { wrapper } = setup();

    wrapper.setState({ last: 2 });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().state.last).toBe(2);
  });

  it('should update state when component receive reviews and pagination as props', () => {
    const { wrapper } = setup();

    wrapper.setProps({
      reviews: allReviews.reviews,
      pagination: allReviews.pagination
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().props.reviews.length).toBe(allReviews.reviews.length);
  });

  describe('handlePageClick', () => {
    it('should update currentpage state when called', () => {
      const { wrapper } = setup();
      const handlePageClickSpy = jest.spyOn(wrapper.instance(), 'handlePageClick');
      wrapper.instance().handlePageClick({ selected: 1 });
      expect(wrapper.instance().state.currentPage).toBe(2);
      expect(handlePageClickSpy).toHaveBeenCalledTimes(1);
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with props required by component from store', () => {
      const store = {
        recipeReviews: {
          reviews: allReviews.reviews,
          pagination: allReviews.pagination
        },
        auth: { currentUser: user },
        isFetching: false
      };

      const {
        reviews,
        pagination,
        currentUser,
        isFetching
      } = mapStateToProps(store);

      expect(reviews.length).toBe(allReviews.reviews.length);
      expect(pagination.last).toBe(allReviews.pagination.last);
      expect(currentUser.fullname).toBe(user.fullname);
      expect(isFetching).toBe(false);
    });
  });
});
