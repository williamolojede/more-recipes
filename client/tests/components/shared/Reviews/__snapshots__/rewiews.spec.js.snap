// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews Component should render a no-reviews page if there are no reviews 1`] = `
<div
  className="recipe__reviews row"
>
  <h3>
    Reviews
  </h3>
  <AddReviewForm
    currentUser={
      Object {
        "fullname": "John Doe",
        "id": 1,
      }
    }
    dispatch={[Function]}
    limit={6}
    recipeId={2}
  />
  <ul
    className="review__list"
  >
    <li
      className="no-reviews"
    >
      eyah 
      <span
        aria-label="sad"
        role="img"
      >
        ðŸ˜¢  
      </span>
       no reviews yet
    </li>
  </ul>
</div>
`;

exports[`Reviews Component should render pagination links if last page is greater than one 1`] = `
<div
  className="recipe__reviews row"
>
  <h3>
    Reviews
  </h3>
  <AddReviewForm
    currentUser={
      Object {
        "fullname": "John Doe",
        "id": 1,
      }
    }
    dispatch={[Function]}
    limit={6}
    recipeId={2}
  />
  <ul
    className="review__list"
  >
    <Review
      key="0"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "i",
          "createdAt": "2018-01-15T01:15:46.499Z",
          "id": 162,
        }
      }
    />
    <Review
      key="1"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "kilode",
          "createdAt": "2018-01-15T01:12:49.228Z",
          "id": 161,
        }
      }
    />
    <Review
      key="2"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "no",
          "createdAt": "2018-01-15T01:04:33.114Z",
          "id": 160,
        }
      }
    />
    <Review
      key="3"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "trefqbiupavjnk;r edwjsklnvre;fwiblknv r;efjkbwnv sr4uefjkwbn bvrknefdw ;svrknefwbkletfdnbsn;tjklrfnwd;sbklnefwdsbtefbwtv4refbwtewhb4trefwhg5reag54trefwgs5rewgs5tr4fewg54trewgh5t",
          "createdAt": "2018-01-15T01:04:13.717Z",
          "id": 159,
        }
      }
    />
    <Review
      key="4"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "b",
          "createdAt": "2018-01-15T00:30:09.398Z",
          "id": 151,
        }
      }
    />
    <Review
      key="5"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "no",
          "createdAt": "2018-01-14T23:53:57.316Z",
          "id": 150,
        }
      }
    />
    <Review
      key="6"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "hi",
          "createdAt": "2018-01-14T23:53:38.073Z",
          "id": 149,
        }
      }
    />
    <Review
      key="7"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "ki lo so",
          "createdAt": "2018-01-14T23:24:37.770Z",
          "id": 148,
        }
      }
    />
    <Review
      key="8"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "what say you",
          "createdAt": "2018-01-14T23:24:24.406Z",
          "id": 147,
        }
      }
    />
    <Review
      key="9"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "nope",
          "createdAt": "2018-01-13T20:45:18.432Z",
          "id": 145,
        }
      }
    />
    <Review
      key="10"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "dgffgd",
          "createdAt": "2018-01-13T20:32:33.124Z",
          "id": 142,
        }
      }
    />
    <Review
      key="11"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "dd",
          "createdAt": "2018-01-13T20:32:30.688Z",
          "id": 141,
        }
      }
    />
  </ul>
  <PaginationBoxView
    activeClassName="active"
    breakLabel="..."
    containerClassName="pagination"
    disableInitialCallback={false}
    disabledClassName="disabled"
    marginPagesDisplayed={3}
    nextClassName="next"
    nextLabel={
      <i
        className="material-icons"
      >
        chevron_right
      </i>
    }
    onPageChange={[Function]}
    pageClassName="waves-effect"
    pageCount={7}
    pageRangeDisplayed={2}
    previousClassName="previous"
    previousLabel={
      <i
        className="material-icons"
      >
        chevron_left
      </i>
    }
  />
</div>
`;

exports[`Reviews Component should render with the right amount of elements 1`] = `
<div
  className="recipe__reviews row"
>
  <h3>
    Reviews
  </h3>
  <AddReviewForm
    currentUser={
      Object {
        "fullname": "John Doe",
        "id": 1,
      }
    }
    dispatch={[Function]}
    limit={6}
    recipeId={2}
  />
  <ul
    className="review__list"
  >
    <Review
      key="0"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "i",
          "createdAt": "2018-01-15T01:15:46.499Z",
          "id": 162,
        }
      }
    />
    <Review
      key="1"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "kilode",
          "createdAt": "2018-01-15T01:12:49.228Z",
          "id": 161,
        }
      }
    />
    <Review
      key="2"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "no",
          "createdAt": "2018-01-15T01:04:33.114Z",
          "id": 160,
        }
      }
    />
    <Review
      key="3"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "trefqbiupavjnk;r edwjsklnvre;fwiblknv r;efjkbwnv sr4uefjkwbn bvrknefdw ;svrknefwbkletfdnbsn;tjklrfnwd;sbklnefwdsbtefbwtv4refbwtewhb4trefwhg5reag54trefwgs5rewgs5tr4fewg54trewgh5t",
          "createdAt": "2018-01-15T01:04:13.717Z",
          "id": 159,
        }
      }
    />
    <Review
      key="4"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "b",
          "createdAt": "2018-01-15T00:30:09.398Z",
          "id": 151,
        }
      }
    />
    <Review
      key="5"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "no",
          "createdAt": "2018-01-14T23:53:57.316Z",
          "id": 150,
        }
      }
    />
    <Review
      key="6"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "hi",
          "createdAt": "2018-01-14T23:53:38.073Z",
          "id": 149,
        }
      }
    />
    <Review
      key="7"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "ki lo so",
          "createdAt": "2018-01-14T23:24:37.770Z",
          "id": 148,
        }
      }
    />
    <Review
      key="8"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "what say you",
          "createdAt": "2018-01-14T23:24:24.406Z",
          "id": 147,
        }
      }
    />
    <Review
      key="9"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "nope",
          "createdAt": "2018-01-13T20:45:18.432Z",
          "id": 145,
        }
      }
    />
    <Review
      key="10"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "dgffgd",
          "createdAt": "2018-01-13T20:32:33.124Z",
          "id": 142,
        }
      }
    />
    <Review
      key="11"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "dd",
          "createdAt": "2018-01-13T20:32:30.688Z",
          "id": 141,
        }
      }
    />
  </ul>
</div>
`;

exports[`Reviews Component should update state when component receive reviews and pagination as props 1`] = `
<div
  className="recipe__reviews row"
>
  <h3>
    Reviews
  </h3>
  <AddReviewForm
    currentUser={
      Object {
        "fullname": "John Doe",
        "id": 1,
      }
    }
    dispatch={[Function]}
    limit={6}
    recipeId={2}
  />
  <ul
    className="review__list"
  >
    <Review
      key="0"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "i",
          "createdAt": "2018-01-15T01:15:46.499Z",
          "id": 162,
        }
      }
    />
    <Review
      key="1"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "kilode",
          "createdAt": "2018-01-15T01:12:49.228Z",
          "id": 161,
        }
      }
    />
    <Review
      key="2"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "no",
          "createdAt": "2018-01-15T01:04:33.114Z",
          "id": 160,
        }
      }
    />
    <Review
      key="3"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "trefqbiupavjnk;r edwjsklnvre;fwiblknv r;efjkbwnv sr4uefjkwbn bvrknefdw ;svrknefwbkletfdnbsn;tjklrfnwd;sbklnefwdsbtefbwtv4refbwtewhb4trefwhg5reag54trefwgs5rewgs5tr4fewg54trewgh5t",
          "createdAt": "2018-01-15T01:04:13.717Z",
          "id": 159,
        }
      }
    />
    <Review
      key="4"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "b",
          "createdAt": "2018-01-15T00:30:09.398Z",
          "id": 151,
        }
      }
    />
    <Review
      key="5"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "no",
          "createdAt": "2018-01-14T23:53:57.316Z",
          "id": 150,
        }
      }
    />
    <Review
      key="6"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "hi",
          "createdAt": "2018-01-14T23:53:38.073Z",
          "id": 149,
        }
      }
    />
    <Review
      key="7"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "ki lo so",
          "createdAt": "2018-01-14T23:24:37.770Z",
          "id": 148,
        }
      }
    />
    <Review
      key="8"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "what say you",
          "createdAt": "2018-01-14T23:24:24.406Z",
          "id": 147,
        }
      }
    />
    <Review
      key="9"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "nope",
          "createdAt": "2018-01-13T20:45:18.432Z",
          "id": 145,
        }
      }
    />
    <Review
      key="10"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "dgffgd",
          "createdAt": "2018-01-13T20:32:33.124Z",
          "id": 142,
        }
      }
    />
    <Review
      key="11"
      review={
        Object {
          "User": Object {
            "fullname": "John Bull",
            "id": 8,
            "username": null,
          },
          "content": "dd",
          "createdAt": "2018-01-13T20:32:30.688Z",
          "id": 141,
        }
      }
    />
  </ul>
  <PaginationBoxView
    activeClassName="active"
    breakLabel="..."
    containerClassName="pagination"
    disableInitialCallback={false}
    disabledClassName="disabled"
    marginPagesDisplayed={3}
    nextClassName="next"
    nextLabel={
      <i
        className="material-icons"
      >
        chevron_right
      </i>
    }
    onPageChange={[Function]}
    pageClassName="waves-effect"
    pageCount={7}
    pageRangeDisplayed={2}
    previousClassName="previous"
    previousLabel={
      <i
        className="material-icons"
      >
        chevron_left
      </i>
    }
  />
</div>
`;
